tests/vm_test.go

```diff
@@ -1136,7 +1136,7 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			_, stdErr, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
 			Expect(err).To(HaveOccurred())
 
-			Expect(strings.HasPrefix(stdErr, "Error from server (AlreadyExists): error when creating")).To(BeTrue(), "command should error when creating VM second time")
+			Expect(strings.Contains(stdErr, "Error from server (AlreadyExists): error when creating")).To(BeTrue(), "command should error when creating VM second time")
```

Comment:
Why this change?

Reply:
the way it is now, the warning for the sepc.running deprecation comes first, so the `HasPrefix` fails.

can be fixed if I append the deprecation warning at the end.

Reply:
Never mind, I was completely wrong here, looks like the warnings are being appended to the stderr by the kubectl client, that is why the HasPrefix fails in this test.

code from kubernetes/kubernetes staging/src/k8s.io/client-go/rest/warnings.go:107:
they are handling warnings by writing them to out Writer
```
// HandleWarningHeader prints warnings with code=299 to the configured writer.
func (w *warningWriter) HandleWarningHeader(code int, agent string, message string) {
	if code != 299 || len(message) == 0 {
		return
	}

	w.writtenLock.Lock()
	defer w.writtenLock.Unlock()

	if w.opts.Deduplicate {
		if _, alreadyWritten := w.written[message]; alreadyWritten {
			return
		}
		w.written[message] = struct{}{}
	}
	w.writtenCount++

	if w.opts.Color {
		fmt.Fprintf(w.out, "%sWarning:%s %s\n", yellowColor, resetColor, message)
	} else {
		fmt.Fprintf(w.out, "Warning: %s\n", message)
	}
}
```

here we can see they pass `IOStreams.ErrOut` for the out writer argument.

```
// NewKubectlCommand creates the `kubectl` command and its nested children.
func NewKubectlCommand(o KubectlOptions) *cobra.Command {
	warningHandler := rest.NewWarningWriter(o.IOStreams.ErrOut, rest.WarningWriterOptions{Deduplicate: true, Color: term.AllowsColorOutput(o.IOStreams.ErrOut)})
```
ref kubernetes/kubernetes staging/src/k8s.io/kubectl/pkg/cmd/cmd.go:311


you can find function signature here:
```
// NewWarningWriter returns an implementation of WarningHandler that outputs code 299 warnings to the specified writer.
func NewWarningWriter(out io.Writer, opts WarningWriterOptions) *warningWriter {
	h := &warningWriter{out: out, opts: opts}
	if opts.Deduplicate {
		h.written = map[string]struct{}{}
	}
	return h
}
```
ref: kubernetes/kubernetes staging/src/k8s.io/client-go/rest/warnings.go:93

Reply:
@xpivarc is it ok?

Reply:
sure, before the addition of the warning logic to the web hook:
`"stderr":"Error from server (AlreadyExists): error when creating \"/tmp/ginkgo2513393333/testvmi-txhzt.json\": virtualmachines.kubevirt.io \"testvmi-txhzt\" already exists\n"`

now after addition of warning about running field 

`"stderr":"Warning: spec.running is deprecated, please use spec.runStrategy instead.\nError from server (AlreadyExists): error when creating \"/tmp/ginkgo2513393333/testvmi-txhzt.json\": virtualmachines.kubevirt.io \"testvmi-txhzt\" already exists\n"`

Reply:
This is fine, I was just curios why we had to change it, thanks